tests/unit/base/trained_model_serialization_unittest.cc.jinja2

```diff
@@ -85,30 +113,50 @@ TEST_F(TrainedModelSerializationTest, {{class}})
 
 	CLabels* predictions=machine->apply(test_feats);
 
-	std::string class_name("{{class}}");
-	std::string filename = "shogun-unittest-trained-model-serialization-" + class_name + ".XXXXXX";
-	generate_temp_filename(const_cast<char*>(filename.c_str()));
-
-	CSerializableAsciiFile *file=new CSerializableAsciiFile(filename.c_str(), 'w');
-	bool save_success=machine->save_serializable(file);
-	file->close();
-	SG_FREE(file);
-	ASSERT_TRUE(save_success);
+	std::string filename;
+	ASSERT_TRUE(serialize_machine(machine, filename));
 
-	file=new CSerializableAsciiFile(filename.c_str(), 'r');
 	auto deserialized_machine=new {{class}}();
-	bool load_success=deserialized_machine->load_serializable(file);
-	file->close();
-	SG_FREE(file);
-	ASSERT_TRUE(load_success);
+	ASSERT_TRUE(deserialize_machine(deserialized_machine, filename));
 
 	CLabels* deserialized_predictions=deserialized_machine->apply(test_feats);
 
 	float64_t accuracy=1e-13;
 	ASSERT(predictions->equals(deserialized_predictions, accuracy, true))
 
-	int delete_success=unlink(filename.c_str());
-	ASSERT_EQ(0, delete_success);
+	SG_FREE(machine);
+	SG_FREE(deserialized_machine);
+	SG_FREE(predictions);
+	SG_FREE(deserialized_predictions);
+}
+{%- endmacro %}
+
+{% macro kernel_machine_test(class) -%}
+TEST_F(TrainedModelSerializationTest, KernelMachine_{{class}})
+{
+	auto machine=new {{class}}();
+	load_data(machine->get_machine_problem_type());
+
+	CGaussianKernel *kernel = new CGaussianKernel(2.0);
+	kernel->init(train_feats, train_feats);
+	machine->set_kernel(kernel);
+	machine->set_labels(train_labels);
+
+	bool train_success=machine->train();
+	ASSERT_TRUE(train_success);
+
+	CLabels* predictions=machine->apply(test_feats);
```

Comment:
Sure, the only difference is just calling `store_model_features()` before the serialization, right?
I'll address this in the next PR after merging this one since I have already some work done for GP and Multiclass.